Rewrite Pulumi Deployments main page (#14093)
pulumi/docs: content/docs/pulumi-cloud/deployments/_index.md

- /docs/pulumi-cloud/deployments/platform/
---

Pulumi Deployments is a managed CI/CD platform purpose-built for infrastructure as code. It provides everything you need to securely deploy infrastructure changes across your organization, including managed compute, secure secrets handling, and deep integration with version control systems.

> ## Key Features

Reviewer: ```suggestion
## Key features
```

> ### Managed Infrastructure CI/CD

Reviewer: ```suggestion
### Managed infrastructure CI/CD
```

- **Git Integration**: Automatically preview infrastructure changes on pull requests and deploy on merge with our GitHub integration.
- **Live Preview Environments**: Each pull request can automatically create a [review stack](/docs/pulumi-cloud/deployments/review-stacks) with real infrastructure, letting you validate changes in a production-like environment before merging.
- **Secure by Default**: Integration with [Pulumi ESC](/docs/pulumi-cloud/esc) ensures your secrets and cloud credentials are securely handled.

### Beyond CI/CD

- **Drift Detection**: [Automatically detect](/docs/pulumi-cloud/deployments/drift) when your infrastructure differs from its desired state.
- **Scheduled Operations**: Run any Pulumi operation (up/preview/refresh) on a [schedule](/docs/pulumi-cloud/deployments/schedules).
- **Temporary Infrastructure**: Automatically tear down development or testing environments with [TTL stacks](/docs/pulumi-cloud/deployments/ttl).

> - **Custom Compute**: Run Pulumi operations on your own infrastructure with [customer-managed agents](/docs/pulumi-cloud/deployments/agents).

Reviewer: ```suggestion
- **Custom compute**: Run Pulumi operations on your own infrastructure with [customer-managed agents](/docs/pulumi-cloud/deployments/customer-managed-agents).
```

### Platform Engineering Features

> - **REST API**: Automate infrastructure operations with our [REST API](/docs/pulumi-cloud/deployments/rest-api) to build custom workflows and self-service platforms.

Reviewer: ```suggestion
- **REST API**: Automate infrastructure operations with our [REST API](/docs/pulumi-cloud/deployments/api/) to build custom workflows and self-service platforms.
```

> - **Deployment Settings**: Define all deployment requirements (source code, credentials, environment variables) in a single configuration, either through the UI or declaratively using the [Pulumi Cloud Provider](/docs/providers/pulumiservice).

Reviewer: This link 404s; where did you want it to go?

Author: Fixed.

> - **Deployment Settings**: Define all deployment requirements (source code, credentials, environment variables) in a single configuration, either through the UI or declaratively using the [Pulumi Cloud Provider](/docs/providers/pulumiservice).

Author: ```suggestion
- **Deployment Settings**: Define all deployment requirements (source code, credentials, environment variables) in a single configuration, either through the UI or declaratively using the [Pulumi Cloud Provider](/registry/packages/pulumiservice).
```

- **Multiple Triggers**: Trigger deployments via git push, REST API, UI button, or schedule to support any workflow.

> - **Best Practices Built-in**: Follow our [deployment patterns](/docs/pulumi-cloud/deployments/using) to implement infrastructure automation at scale.

Reviewer: ```suggestion
- **Best Practices Built-in**: Follow our [deployment patterns](/docs/pulumi-cloud/deployments/reference/) to implement infrastructure automation at scale.
```

> ## How It Works

Reviewer: ```suggestion
## How it works
```

Pulumi Deployments combines three key components to enable secure, scalable infrastructure deployments:

1. **Deployment Settings**: Define everything needed to deploy your infrastructure:

   - Source code location and branch
   - Cloud credentials and OIDC configuration
   - Environment variables and secrets
   - Build requirements and custom Docker images

1. **Managed Compute**: Run Pulumi operations on secure, isolated compute instances that:

   - Scale automatically with your deployment needs
   - Handle concurrent deployments across your organization
   - Provide detailed logs and status updates
   - Can be replaced with your own compute resources if needed

1. **Flexible Triggers**: Choose how deployments are initiated:

   - Git push to deploy
   - Pull request preview environments
   - REST API calls
   - UI-triggered deployments
   - Scheduled operations
   - Custom automation via the Remote Automation API

> ## Getting Started

Reviewer: ```suggestion
## Getting started
```

The fastest way to get started with Pulumi Deployments is to:

> 1. Create a new project using our [New Project Wizard](link-to-wizard), which will automatically configure Deployments for your repository

Reviewer 2: Is `link-to-wizard` some sort of internal magic or is this a TODO?

Author: It was a TODO, but I am utterly mesmerized as to why the link works. Shook, in fact.

Maybe it's because the title of the page matches the link text? Anyway, fixing to a more canonical link.

1. Follow our [Getting Started guide](/docs/pulumi-cloud/deployments/get-started) to configure an existing project

> 1. Explore our [deployment patterns](/docs/pulumi-cloud/deployments/using) to implement common infrastructure automation scenarios

Reviewer: ```suggestion
1. Explore our [deployment patterns](/docs/pulumi-cloud/deployments/reference/) to implement common infrastructure automation scenarios
```
